### ISSUE TITLE
Add insight dependency to anonymously report usage metrics

### ISSUE THREAD

**Issue by bmpvieira** [bionode/bionode#34]
We should probably add [yeoman/insight](https://github.com/yeoman/insight) ASAP to all modules with .askPermission() enabled by default. This data could be very useful when applying for future grants.

**Comment by thejmazz** [comment]
Good idea, looking at the readme, I think you create URLs to display info about specific commands?

So like
```javascript
insight.track('ncbi', 'download')
```

and 
```javascript
insight.track('ncbi', 'search')
```

for the NCBI module would count how many times specific commands were called? 

And at a more basic level, keep track of which modules are being installed.

Could also be cool to get CLI vs. require metrics.

Is this something that is triggered after `npm install`? I think you can identify a script to run in `package.json` after install finishes, so it could probably be put there.

**Comment by tiagofilipe12** [comment]
It would help if you have the bionode-ncbi example implementation somehow documented here.

**Comment by bmpvieira** [comment]
Do:
```bash
git clone git@github.com:bionode/bionode-MODULE_NAME.git
cd bionode-MODULE_NAME
npm install --save-dev insight
```
Then add to the lib folder the [anonymous-tracking.js](https://github.com/bionode/bionode-ncbi/blob/master/lib/anonymous-tracking.js) file.

[Require that file](https://github.com/bionode/bionode-ncbi/blob/master/lib/bionode-ncbi.js#L41) in the main module JS file.
Add [tracking to relevant functions](https://github.com/bionode/bionode-ncbi/blob/master/lib/bionode-ncbi.js#L110), i.e., functions that are part of the API and that we want to track usage. The format should be:

```javascript
insight.track('MODULE_NAME_SUFFIX', 'FUNCTION_NAME')
# Example
insight.track('ncbi', 'search')
```

Also add [tracking to CLI](https://github.com/bionode/bionode-ncbi/blob/master/cli.js#L117).
